### ISSUE TITLE
代码疑问

### ISSUE THREAD

**Issue by maxyonghuawei** [HLearning/leetcode#5]
大佬，小弟初学，请教一下代码。
问题一：
96-97行这段代码是为了让原来vgg16前15层权重可以训练的意思吧。
`or index in range(15):
            model.layers[index].trainable = True`
有点看不明白，is_train = False 时加载原来的权重不会执行训练，is_train = True时才会重新训练，上面代码才有效，那反正都是要训练，上面model.layers[index].trainable = True不都是一样的么，难道keras默认会锁住原来vgg16权重不让训练？
问题二：
小弟我从新看了一下vgg16的网络结构，感觉block4_pool应该是第十四层啊，那为什么上面要写range(15)啊？
`block4_pool = vgg16_model.get_layer('block4_pool').output` 
靴靴大佬~

**Comment by HLearning** [comment]
第一个问题， keras不会默认锁住权重不让训练， 如果默认加载imagenet的权重的话， 你又想训练这部分权重的话， 上面的代码可以不写， 如果你想锁住不训练那一部分代码的话， is_train = False。
关于第二个问题， 你可以打印出网络结构看一下, 可以对应看一下层数

**Comment by maxyonghuawei** [comment]
谢谢，大佬快速回复。
第一个问题我看了一下VGG16到block4_pool好像确实是14层啊，难道我还能数错==
![the-structure-of-the-vgg16-net-which-has-six-main-stages-conv-means-convolutional](https://user-images.githubusercontent.com/46740716/51234719-89496700-19a8-11e9-900e-85d8dc100544.png)

**Comment by maxyonghuawei** [comment]
还有一个问题，小弟我从新研究了@zhixuhao大佬的代码，然后又看了Unet原文的结构，感觉最后在输出维度为一维之前应该还有一个输出维度为2的卷积核。
看你代码第84行，跟zhixuhao大佬不同的是又做了一次输出维度为64的卷积层：
`block10_conv1 = Conv2D(64, 3, activation='relu', padding='same', kernel_initializer='he_normal')(block9_conv2)`
而zhixuhao大佬这边则是和原文一样的2维度的输出：
`conv9 = Conv2D(2, 3, activation = 'relu', padding = 'same', kernel_initializer = 'he_normal')(conv9)`
那我如果按大佬你这么改会有什么影响吗？

**Comment by HLearning** [comment]
关于层数的问题， https://keras.io/applications/
```
# let's visualize layer names and layer indices to see how many layers
# we should freeze:
for i, layer in enumerate(base_model.layers):
   print(i, layer.name)
```

第二个问题， 并没有什么影响， 只是你想在这一层用多少神经元的问题， 你可以用2个， 也可以用64个， 还可以选其他的值， 还有这个地方不能说是输出维度为一维还是二维， 这里你可以称他为输出张量的深度

**Comment by wangzhenlincmd** [comment]
博主，如果没有预训练权重，该咋使用呢？
